### ISSUE TITLE
Sort by recent visit in "VISITOR HOSTNAMES AND IPS"

### ISSUE THREAD

**Issue by stormyqq** [allinurl/goaccess#686]
I suggest goaccess developers to consider to implement a feature which allows to sort the data by "date" in "VISITOR HOSTNAMES AND IPS".  with the current implementation, it's not possible to find out the recent visits because it doesn't state the date. but with this new features, users then can sort the data by date. This is quite useful to find out a large number of requests from a particular ip address received to the server lately. for instance, let's say we received 1000 hits today, and on the last week, in the last month? then how we are gonna find out the recent hits without knowing about when they received it? so please implement that in the next version, otherwise the whole "VISITOR HOSTNAMES AND IPS" doesn't do much.

**Comment by allinurl** [comment]
I'm not sure I follow your example. Could you please clarify the use case of this? Thanks.

**Comment by stormyqq** [comment]
something like this in awstats
http://i.imgur.com/95UbcS1.png
as you can see, it is sorted to display latest visits along with the hits it received. so by that way, if the server is receiving a large amount of hits from a particular ip it always shows up on top of the list. so we can block that ip, then again repeat the same thing on other bad ips.

**Comment by allinurl** [comment]
From your screenshot, is it simply recording the date of last visit for each host? or is it doing something else with the last visit field?

**Comment by stormyqq** [comment]
well I meant it's sorted to display last visits. So if the server is receiving a lot of hits, usually that ip always stays top, and therefore we can recognize it so easily. with the current method, it's not possible, because it can only be sorted to "hits" , "visits", "bandwidth', so if you could add another field "date" apart from the existing ones, it would be great.

**Comment by allinurl** [comment]
I can certainly add another column to display the last visit, and as the other columns, I'll add the option to sort by last visit.

Now I'm curious if it's worth adding this field to all panels... any thoughts?

**Comment by stormyqq** [comment]
yeah adding the last visit to other charts is a good idea too, because people may like to see when a page is requested at last. when a static content is requested at last.

**Comment by allinurl** [comment]
I'll look into this. Thanks.

**Comment by jh537** [comment]
I also have big interest. Can one test the function somewhere? I have compiled from GIT, however, I do not find it?

**Comment by allinurl** [comment]
It hasn't been implemented. Still in the works. Stay tuned :)

**Comment by jh537** [comment]
Okay, many thanks for the quick answer. A really great programm!

**Comment by tanvlt** [comment]
Hi allinurl,
How about this feature.
Thanks.

**Comment by LeOSW42** [comment]
Hi there!
I'm using goaccess to monitor a very small web service, and I'm interested to see lasts visits. In my case it's usefull to daily check if my website have been accessed by new humans, and I also check geoIP for lasts visits and if it is a crawler or not. I think could be good to implement the ability to add this columns to the VISITORS panel, with dynamic sorting table, as in #117 
Thanks
